test/e2e/terraform_network_security_group_test.go

```diff
@@ -22,14 +22,23 @@ func TestExamples(t *testing.T) {
 		if strings.HasPrefix(directory.Name(), "_") || !directory.IsDir() {
 			continue
 		}
-		t.Run(directory.Name(), func(t *testing.T) {
-			test_helper.RunE2ETest(t, "../../", fmt.Sprintf("examples/%s", directory.Name()), terraform.Options{
-				Upgrade: true,
-			}, func(t *testing.T, output test_helper.TerraformOutput) {
-				nsgId, ok := output["network_security_group_id"].(string)
-				assert.True(t, ok)
-				assert.Regexp(t, "/subscriptions/.+/resourceGroups/.+/providers/Microsoft.Network/networkSecurityGroups/.+", nsgId)
+		useForEach := []bool{
+			false,
+			true,
+		}
+		for _, f := range useForEach {
+			t.Run(directory.Name(), func(t *testing.T) {
```

Comment:
As we're running all our tests in parallel, would you please add the `userForEach`'s value to sub-test's name like this way?:

```go
t.Run(fmt.Sprintf("%s-%t", directory.Name(), f), func(t *testing.T) {
```